### PR TITLE
switch from winapi-rs to windows-sys

### PR DESCRIPTION
**Pull request by Philosobyte**
`windows-sys` is officially supported by Microsoft and contains many more APIs than `winapi-rs` does. Switching to `windows-sys` makes future work easier.
I started from @sn99's PR #48 and discovered the reason for the test failure. HKEY is defined in `windows-sys` as an isize. It is possible for `HKEY_CLASSES_ROOT` to overflow this isize and break comparisons in the `RegKey::close_(&mut self)` method, thus resulting in RegKeys not being dropped when they go out of scope. The easiest fix I could think of is to cast HKEYs to a usize during the comparison.

With all due respect to @sn99, I've removed most of the stylistic changes so the PR is more focused and easier for @gentoo90 to review.

I'm pretty new to both Windows development and Rust, so I would take any feedback vigorously.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,14 @@ categories = ["api-bindings", "os::windows-apis"]
 
 [dependencies]
 cfg-if = "1.0"
-winapi = { version = "0.3.9", features = ["impl-default", "impl-debug", "minwindef", "minwinbase", "timezoneapi", "winerror", "winnt", "winreg", "handleapi"] }
+windows-sys = {version = "0.45.0", features = [
```
Comment (nickelc) [inline Cargo.toml:16]: `windows-sys` 0.48 was released some days ago.

Comment (Philosobyte) [reply Cargo.toml]: You're right. I was relying on my IDE's autocomplete, which appears to be outdated. I've bumped the version number up. Tests still pass.

**Comment by Philosobyte** [comment]
Hmm, it appears @gentoo90 is working on a branch to switch to windows-rs as we speak. That will be even better - I found myself wanting the types in windows-rs which weren't in windows-sys. I will close this PR and eagerly await for gentoo90's work.

**Comment by gentoo90** [comment]
Rebased onto the latest `master` and published as part of version [`0.50.0`](https://crates.io/crates/winreg/0.50.0)

**Comment by raymundane** [comment]
> Rebased onto the latest `master` and published as part of version [`0.50.0`](https://crates.io/crates/winreg/0.50.0)

Thank you!

**Pull request closed without merging**